app.go

```diff
@@ -678,43 +677,50 @@ func (a *Application) processTx(txBytes []byte, isCheckTx bool) (TxHandlerResult
 	receiptHandler := a.ReceiptHandlerProvider.Store()
 	defer receiptHandler.DiscardCurrentReceipt()
 
-	r, err := a.TxHandler.ProcessTx(state, txBytes, isCheckTx)
+	r, err := a.TxHandler.ProcessTx(state, txBytes, false)
 	if err != nil {
+		log.Error(fmt.Sprintf("DeliverTx: %s", err.Error()))
 		storeTx.Rollback()
-		// TODO: save receipt & hash of failed EVM tx to node-local persistent cache (not app state)
-		return r, err
+		// Pass tx hash generated from go-etheruem back to Tendermint on failed EVM txs
```

Comment:
-> `Pass the EVM tx hash (if any) back to Tendermint so it stores it in block results`

Reply:
Done